src/main/resources/hudson/tasks/junit/History/index.jelly

```diff
@@ -24,41 +24,40 @@ THE SOFTWARE.
 		<j:set var="start" value="${it.asInt(request.getParameter('start'),0)}"/>
 		<j:set var="end" value="${it.asInt(request.getParameter('end'),start+24)}"/>
 		<j:set var="rangeParameters" value="start=${start}&amp;end=${end+1}"/>
-		<script type="text/javascript">
-			function setCount() {
-				document.getElementById("graph").src = "countGraph/png?${rangeParameters}";
-				document.getElementById("graph").lazyMap = "countGraph/map?${rangeParameters}";
-                document.getElementById("graph").alt = "[Count graph]";
-				document.getElementById("duration-link").style.display = "";
-				document.getElementById("count-link").style.display = "none";
-			}
-			function setDuration() {
-				document.getElementById("graph").src = "durationGraph/png?${rangeParameters}"
-				document.getElementById("graph").lazyMap = "durationGraph/map?${rangeParameters}"
-                document.getElementById("graph").alt = "[Duration graph]";
-				document.getElementById("duration-link").style.display = "none";
-				document.getElementById("count-link").style.display = "";
-			}
-		 </script>
+		<st:adjunct includes="io.jenkins.plugins.echarts"/>
 
 		<st:include from="${it.testObject}" it="${it.testObject}" page="sidepanel.jelly" />
 		<l:main-panel>
-		<H2>${%title(it.testObject.displayName)}</H2>
       <j:choose>
         <j:when test="${it.historyAvailable()}">
-          <div align="center">
-            <img id="graph" src="durationGraph/png?${rangeParameters}" width="600" height="300" lazymap="durationGraph/map?${rangeParameters}" alt="[Duration graph]"/>
-          </div>
-          <div align="center">
-          	show
-          	<a id="count-link" href="#" onclick='javascript:setCount()'>count</a>
-          	<a id="duration-link" href="#" onclick="javascript:setDuration()" style="display:none;">duration</a>
-          </div>
+					<bs:card title="${%History}" fontAwesomeIcon="chart-line">
+						<div id="trend-carousel" class="carousel slide" data-interval="false">
+							<div class="carousel-inner">
+								<div class="carousel-item active">
+									<div id="test-result-trend-chart" class="graph-cursor-pointer card-chart"/>
+								</div>
+								<div class="carousel-item">
+									<div id="test-duration-trend-chart" class="graph-cursor-pointer card-chart"/>
+								</div>
+							</div>
+							<a class="carousel-control-prev" data-target="#trend-carousel" role="button" data-slide="prev">
+								<span class="carousel-control-prev-icon" aria-hidden="true"/>
+								<span class="sr-only">Previous</span>
+							</a>
+							<a class="carousel-control-next" data-target="#trend-carousel" role="button" data-slide="next">
+								<span class="carousel-control-next-icon" aria-hidden="true"/>
+								<span class="sr-only">Next</span>
+							</a>
+						</div>
+					</bs:card>
+					
+					<c:trend-chart it="${it}" title="${%title(it.testObject.displayName)}" enableLinks="true"/>
```

Comment:
Is this another trend chart? That one loads the JS libs as well. Maybe this causes an ordering problem?

Reply:
This trend chart actually is intended to be used on the build page only...

Reply:
ah sure no I just left that in it's not needed

Reply:
didn't help

Reply:
Hmm... it looks good, but this JS/Jelly errors are so hard to spot...